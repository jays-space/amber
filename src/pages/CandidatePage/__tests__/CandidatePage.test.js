import { act, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { CandidatePage } from "..";
import { BrowserRouter } from "react-router-dom";

const CandidatePageWithMockedRouter = () => {
  return (
    <BrowserRouter>
      <CandidatePage />
    </BrowserRouter>
  );
};

describe("<CandidatePage />", () => {
  it("should render correctly", async () => {
    render(<CandidatePageWithMockedRouter />);

    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(async () => {
      const spinnerComponent = screen.getByTestId("spinner");
      expect(spinnerComponent).toBeInTheDocument();
    });

    await act(async () => {
      const pageComponent = await screen.findByTestId("candidate-page");
      expect(pageComponent).toBeInTheDocument();
      expect(pageComponent).toMatchSnapshot();
    });
  });
});
