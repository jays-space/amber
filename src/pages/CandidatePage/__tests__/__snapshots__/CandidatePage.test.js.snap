// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CandidatePage /> should render correctly 1`] = `
<div
  class="flex flex-col justify-center"
  data-testid="candidate-page"
>
  <div
    class="flex flex-col lg:flex-row justify-between items-center my-10 lg:my-0"
    data-testid="candidate-page-header"
  >
    <div
      class="flex flex-col lg:flex-row items-center"
    >
      <div
        class="hidden lg:flex"
      >
        <button
          class="flex flex-row justify-center items-center mx-2 last-of-type:mr-0 py-2 px-6 text-slate-800 text-base font-bold tracking-widest rounded-md capitalize cursor-pointer hover:bg-opacity-75 transform ease-out duration-100 w-11 aspect-square !p-2 bg-slate-500  active:bg-slate-600"
          data-testid="button-icon-dark"
        >
          <svg
            class="text-slate-500 !text-white"
            data-testid="icon-back"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 8a.5.5 0 0 0-.5-.5H2.707l3.147-3.146a.5.5 0 1 0-.708-.708l-4 4a.5.5 0 0 0 0 .708l4 4a.5.5 0 0 0 .708-.708L2.707 8.5H14.5A.5.5 0 0 0 15 8z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <img
        alt="Romana Francois's avatar"
        class="rounded-full mx-4"
        src="https://randomuser.me/api/portraits/women/27.jpg"
      />
      <div
        class="flex flex-col justify-center items-center"
      >
        <div
          class="flex items-center gap-2"
        >
          <h3
            class=" text-slate-700 undefined text-2xl font-bold"
          >
            Romana Francois
          </h3>
          <p
            class="tracking-wide  text-slate-700 capitalize opacity-75"
            data-testid="paragraph"
          >
            (CH)
          </p>
        </div>
        <p
          class="tracking-wide  text-slate-700 capitalize opacity-75"
          data-testid="paragraph"
        >
          dev team ~ fullstack developer
        </p>
        <div
          class="relative z-20"
          title="application-stage"
        >
          <div
            class=" mt-2 flex"
          >
            <div
              class="px-2.5 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-yellow-500"
            >
              <p
                class="tracking-wide  text-slate-700 text-white"
                data-testid="paragraph"
              >
                1
              </p>
            </div>
            <div
              class="px-2.5 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-yellow-500"
            >
              <p
                class="tracking-wide  text-slate-700 text-white"
                data-testid="paragraph"
              >
                2
              </p>
            </div>
            <div
              class="px-2.5 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-yellow-500"
            >
              <p
                class="tracking-wide  text-slate-700 text-white"
                data-testid="paragraph"
              >
                3
              </p>
            </div>
            <div
              class="px-2.5 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-yellow-500"
            >
              <p
                class="tracking-wide  text-slate-700 text-white"
                data-testid="paragraph"
              >
                4
              </p>
            </div>
            <div
              class="px-2.5 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-yellow-500"
            >
              <p
                class="tracking-wide  text-slate-700 text-white"
                data-testid="paragraph"
              >
                5
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="hidden lg:flex"
    >
      <button
        class="flex flex-row justify-center items-center mx-2 last-of-type:mr-0 py-2 px-6 text-slate-800 text-base font-bold tracking-widest rounded-md capitalize cursor-pointer hover:bg-opacity-75 transform ease-out duration-100 w-11 aspect-square !p-2 bg-white hover:bg-slate-50 active:bg-slate-200"
        data-testid="button-icon-light"
      >
        <svg
          class="text-slate-500 text-slate-700"
          data-testid="icon-previous"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.293 6.293 7.586 12l5.707 5.707 1.414-1.414L10.414 12l4.293-4.293z"
          />
        </svg>
      </button>
      <button
        class="flex flex-row justify-center items-center mx-2 last-of-type:mr-0 py-2 px-6 text-slate-800 text-base font-bold tracking-widest rounded-md capitalize cursor-pointer hover:bg-opacity-75 transform ease-out duration-100 w-11 aspect-square !p-2 bg-white hover:bg-slate-50 active:bg-slate-200"
        data-testid="button-icon-light"
      >
        <svg
          class="text-slate-500 text-slate-700"
          data-testid="icon-next"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.707 17.707 16.414 12l-5.707-5.707-1.414 1.414L13.586 12l-4.293 4.293z"
          />
        </svg>
      </button>
      <button
        class="flex flex-row justify-center items-center mx-2 last-of-type:mr-0 py-2 px-6 text-slate-800 text-base font-bold tracking-widest rounded-md capitalize cursor-pointer hover:bg-opacity-75 transform ease-out duration-100 !py-1.5 text-cyan-800 border-2 border-cyan-800 border-opacity-75  active:border-cyan-900"
        data-testid="button-secondary"
      >
        <svg
          class="text-slate-500 mr-2 !text-cyan-800"
          data-testid="icon-message"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 2H8C4.691 2 2 4.691 2 8v13a1 1 0 0 0 1 1h13c3.309 0 6-2.691 6-6V8c0-3.309-2.691-6-6-6zm4 14c0 2.206-1.794 4-4 4H4V8c0-2.206 1.794-4 4-4h8c2.206 0 4 1.794 4 4v8z"
          />
          <path
            d="M7 9h10v2H7zm0 4h7v2H7z"
          />
        </svg>
        send message
      </button>
      <button
        class="flex flex-row justify-center items-center mx-2 last-of-type:mr-0 py-2 px-6 text-slate-800 text-base font-bold tracking-widest rounded-md capitalize cursor-pointer hover:bg-opacity-75 transform ease-out duration-100 !py-1.5 text-cyan-800 border-2 border-cyan-800 border-opacity-75  active:border-cyan-900"
        data-testid="button-secondary"
      >
        <svg
          class="text-slate-500 mr-2 !text-cyan-800"
          data-testid="icon-email"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        send email
      </button>
    </div>
  </div>
  <div
    class="my-6 py-3 px-4 rounded-lg shadow-lg bg-white"
  >
    <div
      class="hidden lg:flex"
      title="tab-menu"
    >
      <div
        class="px-6 py-4 cursor-pointer hover:bg-slate-50 transform ease-out delay-100 border-b-2 border-yellow-500"
      >
        <p
          class="tracking-wide  text-slate-700 font-bold text-sm uppercase"
          data-testid="paragraph"
        >
          details
        </p>
      </div>
      <div
        class="px-6 py-4 cursor-pointer hover:bg-slate-50 transform ease-out delay-100 false"
      >
        <p
          class="tracking-wide  text-slate-700 opacity-80 text-sm uppercase"
          data-testid="paragraph"
        >
          resume
        </p>
      </div>
      <div
        class="px-6 py-4 cursor-pointer hover:bg-slate-50 transform ease-out delay-100 false"
      >
        <p
          class="tracking-wide  text-slate-700 opacity-80 text-sm uppercase"
          data-testid="paragraph"
        >
          hiring pipeline
        </p>
      </div>
      <div
        class="px-6 py-4 cursor-pointer hover:bg-slate-50 transform ease-out delay-100 false"
      >
        <p
          class="tracking-wide  text-slate-700 opacity-80 text-sm uppercase"
          data-testid="paragraph"
        >
          interviews
        </p>
      </div>
    </div>
    <div
      class="mt-5 lg:mt-20 lg:px-4"
    >
      <div
        title="about-application"
      >
        <h4
          class="mb-6 lg:mb-2 text-slate-700 capitalize text-xl font-bold"
        >
          application
        </h4>
        <div
          class="md:p-4 grid grid-cols-1 md:grid-cols-2 gap-6 md:border rounded-md"
        >
          <div
            class="flex flex-col hidden md:grid"
            data-testid="registration-details"
          >
            <p
              class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
              data-testid="paragraph"
            >
              registration
            </p>
            <span
              class="flex"
            >
              <p
                class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                data-testid="paragraph"
              >
                Wed, 28 Sept 2016
              </p>
              <p
                class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                data-testid="paragraph"
              >
                (5 years ago)
              </p>
            </span>
          </div>
          <div
            class="grid gap-6 md:hidden"
          >
            <div
              class="flex flex-col "
              data-testid="registration-details"
            >
              <p
                class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                data-testid="paragraph"
              >
                registration
              </p>
              <span
                class="flex"
              >
                <p
                  class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                  data-testid="paragraph"
                >
                  Wed, 28 Sept 2016
                </p>
              </span>
            </div>
            <div
              class="flex flex-col "
              data-testid="registration-details"
            >
              <p
                class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                data-testid="paragraph"
              >
                registration
              </p>
              <span
                class="flex"
              >
                <p
                  class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                  data-testid="paragraph"
                >
                  (5 years ago)
                </p>
              </span>
            </div>
          </div>
          <div
            class="flex flex-col "
            data-testid="identification-details"
          >
            <p
              class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
              data-testid="paragraph"
            >
              identification
            </p>
            <span
              class="flex"
            >
              <p
                class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                data-testid="paragraph"
              >
                AVS
              </p>
              <p
                class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                data-testid="paragraph"
              >
                - 756.9989.7526.17
              </p>
            </span>
          </div>
        </div>
      </div>
      <div
        class="mt-16 lg:mt-10 mb-2 grid grid-flow-row gap-6"
      >
        <div
          class="grid grid-cols-1 lg:grid-cols-2 gap-6"
        >
          <div
            title="about-candidate"
          >
            <h4
              class="mb-6 md:mb-2 text-slate-700 capitalize text-xl font-bold"
            >
              basic info
            </h4>
            <div
              class="md:p-4 grid grid-rows-1 gap-6 md:border rounded-md"
            >
              <div
                class="flex flex-col "
                data-testid="full name-details"
              >
                <p
                  class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                  data-testid="paragraph"
                >
                  full name
                </p>
                <span
                  class="flex"
                >
                  <p
                    class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                    data-testid="paragraph"
                  >
                    Romana
                  </p>
                  <p
                    class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                    data-testid="paragraph"
                  >
                    Francois
                  </p>
                  <p
                    class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                    data-testid="paragraph"
                  >
                    (Madame)
                  </p>
                </span>
              </div>
              <div
                class="flex flex-col "
                data-testid="gender-details"
              >
                <p
                  class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                  data-testid="paragraph"
                >
                  gender
                </p>
                <span
                  class="flex"
                >
                  <p
                    class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                    data-testid="paragraph"
                  >
                    female
                  </p>
                </span>
              </div>
              <div
                class="flex flex-col "
                data-testid="dob-details"
              >
                <p
                  class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                  data-testid="paragraph"
                >
                  dob
                </p>
                <span
                  class="flex"
                >
                  <p
                    class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                    data-testid="paragraph"
                  >
                    Fri, 24 Apr 1964
                  </p>
                  <p
                    class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                    data-testid="paragraph"
                  >
                    (57)
                  </p>
                </span>
              </div>
              <div
                class="flex flex-col text-sm lg:text-base"
                data-testid="email-details"
              >
                <p
                  class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                  data-testid="paragraph"
                >
                  email
                </p>
                <span
                  class="flex"
                >
                  <p
                    class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 false"
                    data-testid="paragraph"
                  >
                    romana.francois@example.com
                  </p>
                </span>
              </div>
              <div
                class="flex flex-col "
                data-testid="phone-details"
              >
                <p
                  class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                  data-testid="paragraph"
                >
                  phone
                </p>
                <span
                  class="flex"
                >
                  <p
                    class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                    data-testid="paragraph"
                  >
                    078 225 75 50
                  </p>
                </span>
              </div>
              <div
                class="flex flex-col "
                data-testid="cell-details"
              >
                <p
                  class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                  data-testid="paragraph"
                >
                  cell
                </p>
                <span
                  class="flex"
                >
                  <p
                    class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                    data-testid="paragraph"
                  >
                    075 437 25 80
                  </p>
                </span>
              </div>
            </div>
          </div>
          <div
            class="mt-10 lg:mt-0"
            title="candidate-address"
          >
            <h4
              class="mb-6 md:mb-2 text-slate-700 capitalize text-xl font-bold"
            >
              address
            </h4>
            <div
              class="grid md:grid-cols-2 gap-6 md:border rounded-md"
            >
              <div
                class="md:p-4 grid grid-cols-1 md:grid-cols-2 lg:grid-cols-1 gap-6"
              >
                <div
                  class="flex flex-col "
                  data-testid="street address-details"
                >
                  <p
                    class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                    data-testid="paragraph"
                  >
                    street address
                  </p>
                  <span
                    class="flex"
                  >
                    <p
                      class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                      data-testid="paragraph"
                    >
                      3471,
                    </p>
                    <p
                      class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                      data-testid="paragraph"
                    >
                      Rue Dubois
                    </p>
                  </span>
                </div>
                <div
                  class="flex flex-col "
                  data-testid="city-details"
                >
                  <p
                    class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                    data-testid="paragraph"
                  >
                    city
                  </p>
                  <span
                    class="flex"
                  >
                    <p
                      class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                      data-testid="paragraph"
                    >
                      Brütten
                    </p>
                  </span>
                </div>
                <div
                  class="flex flex-col "
                  data-testid="state-details"
                >
                  <p
                    class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                    data-testid="paragraph"
                  >
                    state
                  </p>
                  <span
                    class="flex"
                  >
                    <p
                      class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                      data-testid="paragraph"
                    >
                      Fribourg
                    </p>
                  </span>
                </div>
                <div
                  class="flex flex-col "
                  data-testid="country-details"
                >
                  <p
                    class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                    data-testid="paragraph"
                  >
                    country
                  </p>
                  <span
                    class="flex"
                  >
                    <p
                      class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                      data-testid="paragraph"
                    >
                      Switzerland
                    </p>
                  </span>
                </div>
                <div
                  class="flex flex-col "
                  data-testid="postcode-details"
                >
                  <p
                    class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                    data-testid="paragraph"
                  >
                    postcode
                  </p>
                  <span
                    class="flex"
                  >
                    <p
                      class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                      data-testid="paragraph"
                    >
                      6951
                    </p>
                  </span>
                </div>
                <div
                  class="flex flex-col "
                  data-testid="timezone-details"
                >
                  <p
                    class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
                    data-testid="paragraph"
                  >
                    timezone
                  </p>
                  <span
                    class="flex"
                  >
                    <p
                      class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                      data-testid="paragraph"
                    >
                      Almaty, Dhaka, Colombo
                    </p>
                    <p
                      class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
                      data-testid="paragraph"
                    >
                      (+6:00)
                    </p>
                  </span>
                </div>
              </div>
              <div
                class="overflow-hidden"
              >
                <img
                  alt="map of coordinates"
                  class="h-full object-cover object-center rounded-md"
                  src="/map.png"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="mt-6"
          title="about-account"
        >
          <div
            class="md:flex items-center mb-4"
          >
            <button
              class="flex flex-row justify-center items-center mx-2 last-of-type:mr-0 py-2 px-6 text-slate-800 text-base font-bold tracking-widest rounded-md capitalize cursor-pointer hover:bg-opacity-75 transform ease-out duration-100 !py-1.5 text-cyan-800 border-2 border-cyan-800 border-opacity-75  active:border-cyan-900"
              data-testid="button-secondary"
            >
              <svg
                class="text-slate-500 mr-2 !text-cyan-800"
                data-testid="icon-hidden"
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                />
                <path
                  d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                />
              </svg>
              show login info
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
